configure.ac

```diff
@@ -414,6 +414,10 @@ AS_IF([test "$transport_xpmem" = "yes" -o "$transport_cma" = "yes"],
        AC_DEFINE([ENABLE_HARD_POLLING], [1], [Enable hard polling])
       ])
 
+if test "$enable_shr_atomics" = "no" -a "$transport_xpmem" = "yes" -a "$transport_ofi" = "no" -a "$transport_portals4" = "no" ; then
+    AC_DEFINE([USE_SHR_ATOMICS], [1], [If defined, the shared memory layer will perform processor atomics.])
+fi
```

Comment:
We have always treated the above as an invalid build; the configure script should emit a warning any time there isn't a transport selected. That having been said, this seems like a good change. You might want to update the description of the `--enable-shr-atomics` to say "default: auto" rather than "default: disabled".

A minor fix here, the check for `"$enable_shr_atomics" = "no"` is not quite right (see Example 1.1 [here](https://autotools.io/autoconf/arguments.html)). It's counterintuitive, but I think this check should be `"$enable_shr_atomics" != "no"` i.e. "not disabled".